### PR TITLE
gambit-scheme 4.9.4

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

previous attempts:
- #103860
- #92438

### CODE REVIEW

**Comment by chenrui333** [comment]
for 13-arm

```
  error: expected relocatable expression
  error: expected relocatable expression
  2 errors generated.
  *** ERROR IN ##main -- target compilation or link failed while compiling
  "/private/tmp/gerbil-scheme-20230222-40705-10mupxg/gerbil-0.17/lib/std/gener...
  *** ERROR IN #:loop38665, "misc/concurrent-plan.ss"@90.7 -- Failure to build (gxc: "generic/dispatch" "-e" "(include \"~~lib/_gambit#.scm\")") "Compilation error; gsc exit with nonzero status"
  *** ERROR; build failed
```

**Comment by mdbergmann** [comment]
The thing is again stuck.
What's the problem?

**Comment by SMillerDev** [comment]
The output says
```
  *** ERROR IN #:loop38665, "misc/concurrent-plan.ss"@90.7 -- Failure to build (gxc: "generic/dispatch" "-e" "(include \"~~lib/_gambit#.scm\")") "Compilation error; gsc exit with nonzero status"

```

**Comment by ober** [comment]
> for 13-arm/13/12-arm
> 
> ```
>   error: expected relocatable expression
>   error: expected relocatable expression
>   2 errors generated.
>   *** ERROR IN ##main -- target compilation or link failed while compiling
>   "/private/tmp/gerbil-scheme-20230222-40705-10mupxg/gerbil-0.17/lib/std/gener...
>   *** ERROR IN #:loop38665, "misc/concurrent-plan.ss"@90.7 -- Failure to build (gxc: "generic/dispatch" "-e" "(include \"~~lib/_gambit#.scm\")") "Compilation error; gsc exit with nonzero status"
>   *** ERROR; build failed
> ```
> 
> relates to [vyzo/gerbil#669](https://github.com/vyzo/gerbil/issues/669)

How is this Gambit issue related to Gerbil not building on OpenBSD?

**Comment by chenrui333** [comment]
rebased and rerun after #133524

File: Formula/gambit-scheme.rb [hunk 1]
```diff
@@ -42,17 +41,11 @@ def install
     ]
 
     system "./configure", *args
-
-    # Fixed in gambit HEAD, but they haven't cut a release
-    inreplace "config.status" do |s|
-      s.gsub! %r{/usr/local/opt/openssl(?!@1\.1)}, "/usr/local/opt/openssl@3"
-      s.gsub! %r{/usr/local/opt/openssl(@\d(\.\d)?)?}, Formula["openssl@3"].opt_prefix
-    end
```
Comment (carlocab) [inline Formula/gambit-scheme.rb:50]: Might need to keep some of this.

Comment (cho-m) [reply Formula/gambit-scheme.rb:50]: Could be done on `configure` instead..

**Review by carlocab — APPROVED** [review]

File: Formula/gambit-scheme.rb (hunk 1 above)
Comment (ober) [reply Formula/gambit-scheme.rb]: configure --with-openssl=/usr/local/myssl works.

**Pull request merged**